source/fe/fe_dgp.cc

```diff
@@ -216,11 +217,52 @@ FiniteElementDomination::Domination
 FE_DGP<dim, spacedim>::compare_for_face_domination(
   const FiniteElement<dim, spacedim> &fe_other) const
 {
-  // check whether both are discontinuous elements, see the description of
-  // FiniteElementDomination::Domination
-  if (dynamic_cast<const FE_DGP<dim, spacedim> *>(&fe_other) != nullptr)
+  return this->compare_for_domination(fe_other, 1);
+}
```

Comment:
Since we override the `compare_for_face_domination()` on derived classes, and let the old `compare_for_face_domination()` refer to those overrides, I think we can not introduce the `final` flag here. It would be nice though, but we require different behaviour on several of these derived classes.

See also @bangerth's [comment](https://github.com/dealii/dealii/pull/7337#discussion_r225973825).